### DISCUSSION TITLE
Add timezone to tzinfo parsing to pydantic 2.0

### DISCUSSION THREAD

**Discussion by 07pepa** [pydantic/pydantic#4708] [Pydantic V2]
Pydantic 1.x is missing native suport for timezone parsing  (and since this will probably will not be added natively). I am asking to add it to pydantic 2.x (does not have to be in rust) 

hopefully there is something like pytz (but there is issue how to keep database of all timezone up to date without need to recompile/ update rust part)

**Comment by samuelcolvin** [comment]
I've converted this to an issue to discuss, see #4710.
